tests/unit/testplan/runnable/interactive/test_irunner.py

```diff
@@ -161,6 +181,38 @@ def test_run_suite(irunner, sync):
     assert irunner.report["test_1"]["Suite"].passed
 
 
+def test_run_suite_with_failed_setup():
+    target = runnable.TestRunner(name="TestRunner")
+
+    local_runner = LocalRunner()
+    test = multitest.MultiTest(
+        name="MT",
+        suites=[FailedSetupSuite()],
+        test_filter=filtering.Filter(),
+        test_sorter=ordering.NoopSorter(),
+        stdout_style=defaults.STDOUT_STYLE,
+    )
+    local_runner.add(test, test.uid())
+    target.resources.add(local_runner)
+
+    with mock.patch("cheroot.wsgi.Server"), mock.patch(
+            "testplan.runnable.interactive.reloader.ModuleReloader"
+    ) as MockReloader:
```

Comment:
Followed the style of the module here, but I think pytest allows parametrized fixtures, so we could simplify this module at some point quite a lot.